src/System.Device.Gpio/System.Device.Gpio.csproj

```diff
@@ -42,4 +43,54 @@
       <IsWinMDFile>true</IsWinMDFile>
     </Reference>
   </ItemGroup>
+
+  <PropertyGroup>
+    <NotSupportedSourceFile>$(IntermediateOutputPath)$(MSBuildProjectName).notsupported.cs</NotSupportedSourceFile>
+  </PropertyGroup>
+  <Target Name="GenerateNotSupportedAssemblyForNetStandardSource"
```

Comment:
Rather than running this in the runtime-specific project build, I think you can run https://github.com/dotnet/arcade/blob/a1805103791e43031355e11c0d037bca803a9593/src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets#L73 in the outer build and just get the reference assembly built in the rid-specific build.  That way the only thing you need to couple to is the ref assembly path, which I believe you already have.